### PR TITLE
Automatic generation of openapi documentation

### PR DESCRIPTION
**Pull request by BigFab**
Add automatic generation of the openapi documentation

Testing with a release candidate 0.6.1-rc3 is working fine. Both Openapi and Soup documents are generated in S3.

Only drawback which looks like OK to me for now: node projects generate an OPENAPI doc under name _gatekeeper-0.2.1-swagger.json_ where go projects use a slightly different name _shoreline-dblp.0.6.1-rc3-swagger.json_ This may cause problems later if we want to consolidate the different documentations, which does not appear like a strategy for now

### CODE REVIEW

**Comment by BigFab** [comment]
Not sure why the build status is not reflected: https://travis-ci.org/github/mdblp/shoreline/builds/703963644
The build on release was done and openapi documentation was pushed to S3: https://s3.console.aws.amazon.com/s3/buckets/com.diabeloop.yourloops.doc/openapi/shoreline/?region=eu-west-1&tab=overview

File: buildDoc.sh [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+$GOPATH/bin/swag --version
+$GOPATH/bin/swag init --generalInfo shoreline.go --output docs
+
+# When tag is present, openapi doc is renamed before being deployed to S3
+if [ -n "${TRAVIS_TAG:-}" ]; then
+    APP="${TRAVIS_REPO_SLUG#*/}"
+    APP_TAG="${APP}-${TRAVIS_TAG}"
+    mkdir docs/openapi
+    mv docs/swagger.json docs/openapi/${APP_TAG}-swagger.json
+fi
```
Comment (franck-fourel) [inline buildDoc.sh]:
```suggestion
fi

```

File: buildDoc.sh [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  go get -u github.com/swaggo/swag/cmd/swag
+fi
+
+$GOPATH/bin/swag --version
+$GOPATH/bin/swag init --generalInfo shoreline.go --output docs
+
+# When tag is present, openapi doc is renamed before being deployed to S3
+if [ -n "${TRAVIS_TAG:-}" ]; then
+    APP="${TRAVIS_REPO_SLUG#*/}"
+    APP_TAG="${APP}-${TRAVIS_TAG}"
```
Comment (franck-fourel) [inline buildDoc.sh]:
```suggestion
    APP_TAG="${APP}-${TRAVIS_TAG/dblp./}"
```
removing the dblp thing

**Review by franck-fourel — CHANGES_REQUESTED** [review]
a proposal to remove the dblp thing if needed. 
works like a charm on my side.

**Review by franck-fourel — APPROVED** [review]
LGTM

**Pull request merged**